ocaml/xapi/xapi_cluster_host_helpers.ml

```diff
@@ -55,7 +55,15 @@ let get_operation_error ~__context ~self ~op =
 
   current_error
 
+let assert_joined_cluster ~__context ~self = function
+  | `enable | `disable ->
+    (* Cannot enable nor disable without joining cluster *)
+    if not (Db.Cluster_host.get_joined ~__context ~self)
+    then raise Api_errors.(Server_error (cluster_host_not_joined, [ Ref.string_of self ]))
+  | _ -> ()
+
 let assert_operation_valid ~__context ~self ~op =
+  assert_joined_cluster ~__context ~self op;
```

Comment:
This should be part of `get_operation_error`, because that is also used during update of allowed operations below.
I know currently the allowed operations are not updated properly, but lets avoid introducing more bugs :)